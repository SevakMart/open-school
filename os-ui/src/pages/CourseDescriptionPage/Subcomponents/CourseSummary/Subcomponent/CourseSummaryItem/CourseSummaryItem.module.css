.mainContainer {
  margin-left: 48px;
}
.mainContainer .mainContent {
  display: inline-block;
  margin-left: 16px;
}
.mainContainer .mainContent h2 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  color: #5E617B;
  margin: 0 0 8px;
}
.mainContainer .mainContent p {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
  line-height: 24px;
  color: #333941;
  margin: 0;
}/*# sourceMappingURL=CourseSummaryItem.module.css.map */