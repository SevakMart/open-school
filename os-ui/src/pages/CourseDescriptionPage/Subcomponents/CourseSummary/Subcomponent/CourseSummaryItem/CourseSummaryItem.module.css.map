{"version":3,"sources":["CourseSummaryItem.module.scss","CourseSummaryItem.module.css","../../../../../../styles/Mixins/FontMixin.scss","../../../../../../styles/Configs/Fonts.scss","../../../../../../styles/Configs/Colors.scss"],"names":[],"mappings":"AAKA;EACI,iBAAA;ACJJ;ADKI;EACI,qBAAA;EACA,iBAAA;ACHR;ADIQ;EETN,yBCDY;EDEZ,kBFS2C;EER3C,gBFQkD;EEPlD,eFOsD;EENtD,iBFM2D;EACjD,cIVM;EJWN,eAAA;ACEZ;ADAQ;EEdN,yBCDY;EDEZ,kBFc2C;EEb3C,gBFakD;EEZlD,eFYsD;EEXtD,iBFW2D;EACjD,cIZO;EJaP,SAAA;ACMZ","file":"CourseSummaryItem.module.css"}