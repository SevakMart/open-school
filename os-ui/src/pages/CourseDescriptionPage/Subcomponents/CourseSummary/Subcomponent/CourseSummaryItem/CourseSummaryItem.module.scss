@import "../../../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../../../styles/Mixins/FontMixin.scss";
@import "../../../../../../styles/Configs/Colors.scss";
@import "../../../../../../styles/Configs/Fonts.scss";

.mainContainer {
    margin-left:48px;

    &::after {
        content: "";
        display: table;
        clear: both;
    }

    .ellipseIcon {
        margin-top: 6px;
    }

    .mainContent {
        display:inline-block;
        margin-left:16px;

        h2{
            @include fontStyle($primary-font,normal,400,14px, normal);
            color:$main-text-color;
            margin:0 0 8px;
        }

        p {
            @include fontStyle($primary-font,normal,600,20px, normal);
            color:$black-text-color;
            margin:0;
        }
    }
}
