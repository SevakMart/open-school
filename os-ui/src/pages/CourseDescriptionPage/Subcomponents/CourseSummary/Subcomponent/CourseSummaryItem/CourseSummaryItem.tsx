import CourseSummaryEllipseIcon from '../../../../../../assets/svg/Course-Summary-Ellipse.svg';
import styles from './CourseSummaryItem.module.scss';

const CourseSummaryItem = ({ title, value }:{title:string, value:number|string}) => {
  const {
    mainContainer, mainContent, ellipseIcon,
  } = styles;

  const formattedTitle = title === 'CourseLevel'
    ? 'Course Level'
    : title.replace(/([A-Z])/g, ' $1').trim();

  return (
    <div className={mainContainer}>
      <img src={CourseSummaryEllipseIcon} className={ellipseIcon} alt="Ellipse" />
      <div className={mainContent}>
        <h2>{formattedTitle}</h2>
        <p>{value}</p>
      </div>
    </div>
  );
};

export default CourseSummaryItem;
