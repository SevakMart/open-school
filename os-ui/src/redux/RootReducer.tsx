import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
