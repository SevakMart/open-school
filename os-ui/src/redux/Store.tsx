import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
