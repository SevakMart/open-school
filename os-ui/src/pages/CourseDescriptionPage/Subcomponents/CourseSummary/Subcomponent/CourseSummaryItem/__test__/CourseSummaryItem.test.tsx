import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../../../redux/Store';
import CourseSummaryItem from '../CourseSummaryItem';

describe('Create test cases for CourseSummaryItem', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <CourseSummaryItem title="Rating" value={5} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
