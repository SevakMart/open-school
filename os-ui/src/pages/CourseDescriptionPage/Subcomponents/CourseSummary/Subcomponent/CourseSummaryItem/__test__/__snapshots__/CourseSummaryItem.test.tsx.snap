// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for CourseSummaryItem Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <img
      alt="Ellipse"
      class="ellipseIcon"
      src="Course-Summary-Ellipse.svg"
    />
    <div
      class="mainContent"
    >
      <h2>
        Rating
      </h2>
      <p>
        5
      </p>
    </div>
  </div>
</DocumentFragment>
`;
